tests/manage/test_dynamic_pvc_accessmodes_with_reclaim_policies.py

```diff
@@ -0,0 +1,375 @@
+        set to Delete/Retain
+        Steps:
+        1. Create Storage Class with reclaimPolicy: Delete/Retain
+        2. Create PVC with 'accessModes: ReadWriteOnce'
+        3. Create two pods using same PVC
+        4. Run IO on first pod
+        5. Verify that second pod is not in Running state
+        6. Delete first pod
+        7. Wait for second pod to be in Running state
+        8. Verify usage of volume in second pod is matching with usage in
+           first pod
+        9. Run IO on second pod
+        10. Delete second pod
+        11. Delete PVC
+        12. Verify PV associated with deleted PVC is also deleted/released
+        """
+        # Create Storage Class with reclaimPolicy: Delete
+        logger.info(
+            f"Creating Storage Class with reclaimPolicy: {reclaim_policy}"
+        )
+        if interface_type == constants.CEPHBLOCKPOOL:
+            secret_name = self.rbd_secret_obj.name
+            interface_name = self.cbp_obj.name
+
+        elif interface_type == constants.CEPHFILESYSTEM:
+            secret_name = self.cephfs_secret_obj.name
+            interface_name = helpers.get_cephfs_data_pool_name()
+
+        sc_obj = helpers.create_storage_class(
+            interface_type=interface_type,
+            interface_name=interface_name,
+            secret_name=secret_name,
+            reclaim_policy=reclaim_policy
+        )
+
+        # Create PVC with 'accessModes' 'ReadWriteOnce'
+        logger.info('Creating PVC with accessModes: ReadWriteOnce')
+        pvc_data = templating.load_yaml_to_dict(constants.CSI_PVC_YAML)
+        pvc_data['metadata']['name'] = helpers.create_unique_resource_name(
+            'test', 'pvc'
+        )
+        pvc_data['metadata']['namespace'] = self.namespace
+        pvc_data['spec']['storageClassName'] = sc_obj.name
+        pvc_data['spec']['accessModes'] = ['ReadWriteOnce']
+        pvc_obj = PVC(**pvc_data)
+        pvc_obj.create()
+
+        # Create two pods using same PVC
+        logger.info(f"Creating two pods using same PVC {pvc_obj.name}")
+
+        # Create first pod
+        if interface_type == constants.CEPHBLOCKPOOL:
+            pod_data = templating.load_yaml_to_dict(constants.CSI_RBD_POD_YAML)
+        elif interface_type == constants.CEPHFILESYSTEM:
+            pod_data = templating.load_yaml_to_dict(
+                constants.CSI_CEPHFS_POD_YAML
+            )
+
+        pod_data['metadata']['name'] = helpers.create_unique_resource_name(
+            'test', 'pod'
+        )
+        pod_data['metadata']['namespace'] = self.namespace
+        pod_data['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = (
+            pvc_obj.name
+        )
+        pod_obj = helpers.create_pod(**pod_data)
+
+        node_pod1 = pod_obj.get()['spec']['nodeName']
+
+        # Create second pod
+        # Try creating pod until it is on a different node than first pod
+        for retry in range(1, 6):
```

Comment:
Looks like this loop is creating a pod in all nodes until two pods are on same node. Instead of this can you please use labels and try creating the node on a specific node? please check.

This may help https://kubernetes.io/docs/tasks/configure-pod-container/assign-pods-nodes/

Reply:
@hnallurv do you want us to use node selector and label selector ? Do we know the exact implementation in OCS 4.2 ? I  have asked Humble for some e.g. as to how they are handling label selectors so as to get the supported steps. 

Please suggest

Reply:
@nehaberry let's try to get that info from Humble asap

Reply:
Can we file an issue here, I think its important to have this supported in the library, basically we want 2 diffferent worker nodes here, selector might help, but the ocp object as I see it doesn't support it.

Reply:
I have filed it: https://github.com/red-hat-storage/ocs-ci/issues/376

Reply:
Second pod will be created on a different node now, modified helper.py for specifying the nodeName